code/mob/living/intangible/flock/flocktrace.dm

```diff
@@ -13,13 +13,16 @@
 /mob/living/intangible/flock/trace/New(atom/loc, datum/flock/F)
 	..()
 
+	src?.abilityHolder = new /datum/abilityHolder/flockmind(src)
+
 	src.real_name = "[pick(consonants_upper)][pick(vowels_lower)].[pick(vowels_lower)]"
 
 	if(istype(F))
 		src.flock = F
 		src.flock.addTrace(src)
 	else
 		src.death() // f u
+	src?.abilityHolder?.addAbility(/datum/targetable/flockmindAbility/createStructure)
```

Comment:
would've runtimed before this point if no src, also didn't you literally just make abilityholder